### PR TITLE
WIP Crutches.Option module

### PR DESCRIPTION
**Pull request by duijf**
Functions to turn this:

``` elixir
@function_name [
  valid_options: [:some, :option],
  default_options: [
    some: "foo"
    option: "bar"
  ]
]
def function_name(args, opts \\ @function_name[:default_options])
def function_name(args, opts) do
  Crutches.Keyword.validate_keys!(opts, @function_name[:valid_options])
  options = Keyword.merge(@function_name[:default_options], opts)

  ...
end
```

Into this:

``` elixir
@function_name [
  valid: [:some, :option],
  defaults: [
    some: "foo",
    option: "bar"
  ]
]
def function_name(args, opts \\ @function_name[:defaults])
def function_name(args, opts) do
  # This validates and merges the options, throwing on error.
  opts = Crutches.Options.combine!(opts, @function_name)

  ...
end
```

For more context, see #42.

Refactoring of the library to make use of these new functions is not going to happen in this PR, in order for everyone to be able to benefit from these functions as quickly as possible.

**TODOs:**
- [x] `Option.combine!(options, configuration)` &mdash; throwing
- [x] `Option.combine(options, configuration)` &mdash; non-throwing
- [x] `Option.combine!(options, configuration, combinator)` &mdash; throwing, with custom combinator.
- [x] `Option.combine(options, configuration, combinator)` &mdash; non-throwing, with custom combinator.
- [x] Move the functions from `Crutches.Keyword` into `Crutches.Option`

### CODE REVIEW

**Comment by duijf** [comment]
Hmm, weird that the build is failing. Might have to do with other warnings, but I doubt that'd make it segfault. Temporarily fixing all warnings (will drop from commit history if unsuccessful).

**Comment by duijf** [comment]
That seems to have done it. If wanted, that commit could go somewhere else.

File: lib/crutches/option.ex [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      {:ok, [foo: "another", bar: "value"]}
+
+      iex> config = [valid: ~w(bar baz)a, defaults: [bar: "good", baz: "values"]]
+      iex> Option.combine([boom: "this blows up"], config)
+      {:error, [:boom]}
+
+  """
+  @spec combine(list(any), list(any)) :: list(any)
+  def combine(opts, config) do
+    Crutches.Option.combine(opts, config, &Elixir.Keyword.merge(&1, &2))
```
Comment (CyrusNuevoDia) [inline lib/crutches/option.ex]: Just call `combine(args)` instead of `Crutches.Option.combine(args)`. Ditto for other occurrences.

P.S. Great work dude!

**Comment by duijf** [comment]
@knrz Incorporated your feedback. Currently working on the last checkbox.

**Pull request merged**